File: app/src/main/java/djordjeh/architecture/mvp/data/source/TaskDataSource.java
package djordjeh.architecture.mvp.data.source;

import android.support.annotation.NonNull;

import java.util.List;

import djordjeh.architecture.mvp.data.model.Task;
import io.reactivex.Maybe;
import io.reactivex.Observable;
import io.reactivex.Single;

public interface TaskDataSource {
    Observable<List<Task>> tasks(boolean forceUpdate);
    Maybe<Task> task(long taskId);
    Single<Task> save(@NonNull Task task);
    Single<Task> delete(@NonNull Task task);
}

File: app/src/main/java/djordjeh/architecture/mvp/data/source/TaskDataSource.kt
package djordjeh.architecture.mvp.data.source

import djordjeh.architecture.mvp.data.model.Task
import io.reactivex.Maybe
import io.reactivex.Observable
import io.reactivex.Single

interface TaskDataSource {
    fun tasks(forceUpdate: Boolean): Observable<List<Task>>
    fun task(taskId: Long): Maybe<Task>
    fun save(task: Task): Single<Task>
    fun delete(task: Task): Single<Task>
}
